qa/test_framework/util.py

```diff
@@ -439,3 +440,46 @@ def get_coinbase_address(node, expected_utxos=None):
     addrs = [a for a in set(addrs) if addrs.count(a) == expected_utxos]
     assert(len(addrs) > 0)
     return addrs[0]
+
+"""
+Perform SC creation, mine mainchain blocks, create genesis info.
+Parameters:
+ - sidechain_id: id of the sidechain to be created
+ - mainchain_node: the mainchain node
+ - secrets: a JSON array of genesis secrets in the following form: [{"secret":"a string", "publicKey":"a string"}...]
+ - amounts: an array of amounts for each secret.
+ 
+Output: an array of two information:
+ - the genesis info used for start the sidechain node
+ - the height of the mainchain block at which the sidechain has been created (useful for future checks of mainchain block reference inclusion)
+
+"""
+def get_genesis_info(sidechain_id, mainchain_node, secrets, amounts):
+    number_of_blocks_to_enable_sc_logic = 219
+    number_of_blocks = mainchain_node.getblockcount()
+    diff = number_of_blocks_to_enable_sc_logic - number_of_blocks
+    if diff > 1:
+        mainchain_node.generate(diff)
+
+    withdrawal_epoch_length = 1000
```

Comment:
make it as an argument, it can be different for different tests

Reply:
ok

Reply:
Added as argument